File: 2026-07-15-15-28-31/android-aosp-agent/frameworks/base/services/core/java/com/android/server/aiagent/LlmClient.java
/*
 * Copyright (C) 2026 The Android Open Source Project
 *
 * LlmClient — LLM 决策抽象。decide() 给定对话历史 + 可用工具,返回一个 Decision:
 *   - finished=true 带文本(summary)
 *   - 否则带要调用的 tool + 参数 JSON
 * 对应 in-app agent 的 com.fly.agent.llm.LlmClient(原样搬,Java 化)。
 *
 * 真实实现(OpenAI 兼容 / 端侧 NNAPI)后续接入;MVP 用 MockLlmClient 跑通链路。
 */

package com.android.server.aiagent;

import org.json.JSONObject;

import java.util.List;

public interface LlmClient {

    class LlmMessage {
        public final String role;   // "system" | "user" | "assistant" | "tool"
        public final String content;
        public LlmMessage(String role, String content) {
            this.role = role; this.content = content;
        }
    }

    class Decision {
        public boolean finished = false;
        public String text = "";
        public String toolName = "";
        public JSONObject toolArgs = new JSONObject();
    }

    Decision decide(List<LlmMessage> messages, List<Tool> tools);
}

File: 2026-07-15-15-28-31/android-inapp-agent/app/src/main/kotlin/com/fly/agent/llm/LlmClient.kt
package com.fly.agent.llm

/**
 * LLM 抽象。agent 回环只依赖此接口：
 * 传入对话历史 + 工具定义，返回一条 assistant 消息
 *（可能带 content，也可能带 tool_calls）。
 */
interface LlmClient {
    val id: String

    /**
     * @param messages 完整对话历史（含 system/user/tool 角色）
     * @param tools    可用工具定义（模型不支持时忽略即可）
     * @return assistant 消息（content 或 tool_calls 至少其一）
     */
    suspend fun chat(messages: List<ChatMessage>, tools: List<ToolDef>): ChatMessage
}
